### ISSUE TITLE
when PROJECT_TYPE = local, not a git repository any more

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#365]
In GitLab by @ editomas on Jan 29, 2019, 17:09

@dbeltrankyl when we set up an experiment and choose "PROJECT_TYPE = local", in case in local we have a git repository, once this is copied in /esarchive/autosubmit/$expid is not any more a git repository. 

It would be useful to copy over to esarchive also the hidden git files (if this is possible) so that in /esarchive/autosubmit/$expid we can keep track, using git commands, of the committed code changes we have used for that experiment.

@jvegreg @orioljorba @f-macchia

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 7, 2019, 17:07

Hi, 

This was discussed in today mwt meeting and we come to the conclusion that it would be problematic if AS allows to keep a track in local of the git changes globally ( could cause coherence issues and a mess in general for the cooperation in the projects) for now this won't be included.

Maybe @mcastril  can give more details about this.(or explain it better than me)

**Comment by kinow** [comment]
In GitLab by @ editomas on Feb 8, 2019, 10:08

yes @mcastril when you have time, let's talk about this. If you could please clarify this sentence:  "keep a track in local of the git changes globally". I was asking for /esarchive/autosubmit/$expid to be a git repository after the copy from local. I was not asking for it to be synchronized with local after the copy. Happy to clarify it in person (if easier), and then report it here. thanks.

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 8, 2019, 15:20

Hi @ editomas , we can talk about it but in fact I find it easier to write it first because I can structure the reasoning. 

Before asking for a feature, even if it may seem obvious, we prefer that users give some context about why that feature is demanded, because in some situations there's an optional way that fulfills the same purpose. Specially in this case I would ask in which context are you working with local projects and also why do you think it would be useful to transfer the versioning metadata.

PROJECT_TYPE=local is thought for very limited cases, in which you maybe don't have network connection (working inside MN4) or you work inside a virtual machine in a static environment.

Let me also explain in a more detailed way, why in the M&W meeting we thought this should not be implemented:

*  As you know, the workflow proposed by CES is to create your own branch, code your developments, always pushing them to a branch, to later test them in the same experiment or in other ones. Using local projects is not compatible with this workflow as it forgets about Gitlab.

*  In that sense, we recently implemented a restriction to avoid copying experiments with uncommited or non-pushed changes. This request would go against that restriction, because you always could bypass it working with local projects.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 8, 2019, 16:07

Hi @mcastril,

>  In that sense, we recently implemented a restriction to avoid copying experiments with uncommited or non-pushed changes. This request would go against that restriction, because you always could bypass it working with local projects.

Regarding to this, what I did was to check commits and then add a warning but it still copies the experiment, do you want that I modify it to actually restrict the copy?

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 8, 2019, 16:12

Well, even if it's a warning the commits are not being copied to the new experiment because no data is moved between remote experiments. The warning is useful to remind users that they are not actually strictly copying the experiment, and that they are leaving some work behind.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 18, 2019, 16:29

closed

**Comment by kinow** [comment]
In GitLab by @ editomas on Feb 18, 2019, 16:47

@mcastril I had time to comment yet on your answer. First of all, thanks for explaining. 
Since you were asking, the only reason I had to ask for this feature was not to have to create a branch in auto-monarch just to have a different link (compared to the master) to a submodule on which one is working. So, I was cloning in local the auto-monarch master, checking out the submodule branch that I wanted to test and run an experiment taking the code from local. It was annoying not to have in /autosubmit/$expid git information about which commits I was testing (since sometime I run many tests at once). 

I found a way around: I create the experiment taking the master from git and then I checkout in /autosubmit/$expid the submodule develop branch I want to test. For some reason, this did not occur to me in the first place. Hence sorry for the noise.
